sympy/core/equation.py

```diff
@@ -0,0 +1,359 @@
+        return self._binary_op(self, other, lambda a, b: a / b)
+
+    def __rtruediv__(self, other):
+        return self._binary_op(other, self, lambda a, b: a / b)
+
+    def __mod__(self, other):
+        return self._binary_op(self, other, lambda a, b: a % b)
+
+    def __rmod__(self,other):
+        return self._binary_op(other, self, lambda a, b: a % b)
+
+    def __pow__(self, other):
+        return self._binary_op(self, other, lambda a, b: a ** b)
+
+    def __rpow__(self, other):
+        return self._binary_op(other, self, lambda a, b: a ** b)
+
+    def _eval_power(self, other):
+        return self.__pow__(other)
+    
+#####
+# Output helper functions
+#####
+    def __repr__(self):
+        return(str(self.lhs)+self.relop+str(self.rhs))
+
+    def _latex(self,obj,**kwargs):
+        from sympy.printing import latex
+        return(latex(self.lhs)+self.relop+latex(self.rhs))
+
+    def __str__(self):
+        return(self.__repr__())
+
+#####
+# Operation helper functions
+#####
+    def expand(self, *args, **kwargs):
+        return Equation(self.lhs.expand(*args, **kwargs),self.rhs.expand(*args, **kwargs))
+
+    def simplify(self, *args, **kwargs):
+        return self._eval_simplify(*args, **kwargs)
+
+    def _eval_simplify(self, *args, **kwargs):
+        return Equation(self.lhs.simplify(*args, **kwargs),self.rhs.simplify(*args, **kwargs))
+
+    def _eval_factor(self, *args, **kwargs):
+        # TODO: cancel out factors common to both sides.
+        return Equation(self.lhs.factor(*args, **kwargs),self.rhs.factor(*args, **kwargs))
+
+    def factor(self, *args, **kwargs):
+        return self._eval_factor(*args, **kwargs)
+
+    def _eval_collect(self, *args, **kwargs):
+        from sympy.simplify.radsimp import collect
+        return Equation(collect(self.lhs, *args, **kwargs),collect(self.rhs, *args, **kwargs))
+
+    def collect(self, *args, **kwargs):
+        return self._eval_collect(*args, **kwargs)
+
+    def evalf(self, *args, **kwargs):
+        return Equation(self.lhs.evalf(*args, **kwargs),self.rhs.evalf(*args, **kwargs))
+
+    def _eval_derivative(self, *args, **kwargs):
+        # TODO Find why diff and Derivative do not appear to pass through kwargs to this.
+        # Since we cannot set evaluation of lhs manually try to be intelligent about when
+        # to do it.
+        from sympy.core.function import Derivative
+        eval_lhs = False
+        if not(isinstance(self.lhs,Derivative)):
+            for sym in args:
+                if sym in self.lhs.free_symbols and not(_sympify(sym).is_number):
+                    eval_lhs=True
+        return Equation(self.lhs.diff(*args, **kwargs, evaluate=eval_lhs), self.rhs.diff(*args, **kwargs))
+
+    def integ(self, *args, **kwargs):
+        """
+        This function is a convenience function that returns a new equation consisting of an unevaluated
+        integral of the lhs as the new lhs and the result of integrating the rhs as the new rhs.
```

Comment:
This seems quite specific. I think it would be better to a method called integrate having the same signature and analogous behaviour as Expr.integrate.

There should be a way to create an unevaluate integral of each side. I don't see why we need a specific way to create an unevaluated integral of one side and an evaluate integral of the other.

Reply:
This resulted from my inability to pass `evaluate=False` through `integrate` and because somehow calls to `Integral` that are nested any deeper seem to evaluate as well. I agree this is very specific, but I got tired of having to write: `Eqn(Integral(t.lhs),integrate(t.rhs))`, when I wanted that output. I would like to simply be able to do something like `integrate(t, evaluate=lhs)`. I suppose we could use another keyword, I was able to get others to pass through.

Reply:
Perhaps a more general method would be a way to apply a function to both sides and call doit on the rhs. 

Although I also am not sure if we really need that specific functionality. It's useful for generating things like integration tables, but Is that the primary thing people will use Equation for? 

You can always do `eq.applyfunc(lambda s: Integral(s, x)).applyright(lambda s.s.doit())`. The lambdas get a bit tiring when dealing with eq, but I'm not sure how to deal with that generally. One suggestion would be for every method on eq to take a `side` argument that defaults to `None` meaning both but can be `'lhs'` or `'rhs'`. It looks like you already have this for a few of the methods.

Reply:
Removed. I have some ideas to make something more general.